src/array_api_stubs/_draft/info.py

```diff
@@ -0,0 +1,163 @@
+      info.capabilities()
+      info.devices()
+      info.dtypes()
+      info.default_dtypes()
+      # ...
+    """
+
+
+def capabilities() -> Capabilities:
+    """
+    Returns a dictionary of array library capabilities.
+
+    The dictionary should contain the following keys:
+
+    -   **boolean indexing**: boolean indicating whether an array library supports boolean indexing.
+    -   **data-dependent shapes**: boolean indicating whether an array library supports data-dependent output shapes.
+
+    Returns
+    -------
+    out: Capabilities
+        a dictionary of array library capabilities.
+    """
+
+
+def default_device() -> device:
+    """
+    Returns the default device.
+
+    Returns
+    -------
+    out: device
+        an object corresponding to the default device.
+    """
+
+
+def default_dtypes(
+    *,
+    device: Optional[device] = None,
+) -> DefaultDataTypes:
+    """
+    Returns a dictionary containing default data types.
+
+    The dictionary should have the following keys:
+
+    -   **real floating**: default real floating-point data type.
+    -   **complex floating**: default complex floating-point data type.
+    -   **integral**: default integral data type.
+    -   **indexing**: default index data type.
+
+    Parameters
+    ----------
+    device: Optional[device]
+        device for which to return default data types. If ``device`` is ``None``, the returned data types must be the default data types for the current device; otherwise, the returned data types must be default data types specific to the specified device. Default: ``None``.
+
+        .. note::
+           Some array libraries have the concept of a device context manager, allowing library consumers to manage the current device context. When ``device`` is ``None``, libraries supporting a device context should return the default data types for the current device. For libraries without a context manager or supporting only a single device, those libraries should return the default data types for the default device.
+
+    Returns
+    -------
+    out: DefaultDataTypes
+        a dictionary containing the default data type for respective data type kinds.
+    """
+
+
+def dtypes(
+    *,
+    device: Optional[device] = None,
+    kind: Optional[Union[str, Tuple[str, ...]]] = None,
+) -> DataTypes:
+    """
+    Returns a dictionary of supported *Array API* data types.
+
+    .. note::
+       While specification-conforming array libraries may support additional data types which are not present in this specification, data types which are not present in this specification should not be included in the returned dictionary.
+
+    .. note::
+       Specification-conforming array libraries must only return supported data types having expected properties as described in :ref:`data-types`. For example, if a library decides to alias ``float32`` as ``float64``, that library must not include ``float64`` in the dictionary of supported data types.
```

Comment:
Most relevant for JAX I think, so pinging @jakevdp for visibility. I think it'll actually be useful to be able to write portable code also for JAX in SciPy & co, while being fully backwards-compatible for JAX users since the current `float64` aliasing to `float32` is unlikely to change. WDYT @jakevdp?

Reply:
^^^^^ @jakevdp (in case you missed the ping) 🙂

Reply:
If returning `float64` in certain cases is a requirement for the array API, then it sounds like JAX is only fully-compatible with the Array API when `JAX_ENABLE_X64=True`. That seems fine, I think.

Reply:
We are considering relaxing dtype requirements, as this is one of the main friction points among array libraries. Part of the impetus for the inspection API is to allow dynamically querying dtype support, and we'll revisit mandatory dtype support in 2024.

The main point of the referenced comment is that array libraries are not allowed to alias dtypes in the list of returned dtypes which don't actually correspond to the underlying storage. E.g., exposing a "float64" dtype, where in reality it is "float32". 

I think we are good here, so I'll go ahead and mark as resolved.